stubs/netaddr/netaddr/eui/ieee.pyi

```diff
@@ -0,0 +1,33 @@
+import _csv
+from _typeshed import StrOrBytesPath
+from collections.abc import Iterable
+from typing import Any, BinaryIO, TextIO
+from typing_extensions import TypeAlias
+
+from netaddr.core import Publisher, Subscriber
+
+_INDEX: TypeAlias = dict[int, list[tuple[int, int]]]
+OUI_INDEX: _INDEX
+IAB_INDEX: _INDEX
+
+class FileIndexer(Subscriber):
+    writer: _csv._writer
+    def __init__(self, index_file: TextIO | StrOrBytesPath | int) -> None: ...
+    def update(self, data: Iterable[Any]) -> None: ...
+
+class OUIIndexParser(Publisher):
+    fh: BinaryIO
+    def __init__(self, ieee_file: BinaryIO | StrOrBytesPath | int) -> None: ...
+    def parse(self) -> None: ...
+
+class IABIndexParser(Publisher):
+    fh: BinaryIO
+    def __init__(self, ieee_file: BinaryIO | StrOrBytesPath | int) -> None: ...
+    def parse(self) -> None: ...
+
+def create_index_from_registry(
+    registry_fh: BinaryIO | StrOrBytesPath | int, index_path: StrOrBytesPath, parser: type[OUIIndexParser] | type[IABIndexParser]
+) -> None: ...
+def create_indices() -> None: ...
+def load_index(index: _INDEX, fp: BinaryIO) -> None: ...
```

Comment:
I agree with the concept of permissive types. A note here though, the docstring says *"Load index from file into index data structure."*. doesn't this mean they expect a file and not any iterable of bytes? The implementation does accept any iterable of bytes `[x.decode('UTF-8') for x in fp]` but I supposed this has to do with some python2 compat.

Reply:
Yep, I saw that comment in the source code too. The trouble is that the concept of a "file-like object" is really very poorly defined in Python. PEP 484 attempted to define "file-like" by adding the `IO` and `BinaryIO` interfaces to the `typing` module. But these interfaces are very hard to satisfy in real life, because they're `ABC`s with a large number of methods. In the wild, people often use custom file-like classes that don't inherit from any of the stdlib file-like classes, and so aren't recognised by type checkers as being subtypes of `BinaryIO`.

If we accept that there's no good way in the current type system of expressing "An object representing an opened file which, when iterated over, yields `bytes` objects" -- then we're left with having to decide these things on a case-by-case basis. We tend to prefer false negatives over false positives in typeshed, so here I'd go with `Iterable[bytes]`.

If it makes you feel better about things, you could define a type alias, and use that in the annotation instead?

```python
# Expected to be an object representing an opened file,
# but the only restrictions at runtime are that it has to be an Iterable of bytes
_BytesFile: TypeAlias = Iterable[bytes]
```

Reply:
That's a great explanation thank you. I'll commit the suggestion.